### PR TITLE
Provide admission server builds for CPU architectures other than amd64

### PR DESCRIPTION
**Pull request by akankshakumari393**
Fixes - #627 

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by shaneutt — COMMENTED** [review]
/ok-to-test

**Review by shaneutt — CHANGES_REQUESTED** [review]
Most of this looks good, but one important thing it does is add separate images for different architectures. It would seem to me that ideally we would want a multi-arch build as this will have the benefit of allowing container runtimes to pull their version automatically without the image name having to be managed at a higher level. @robscott, @youngnick is there any reason we _wouldn't_ want or be able to do this as a multi-arch build? :thinking:

File: Dockerfile [hunk 1]
```diff
@@ -18,7 +18,8 @@ WORKDIR /go/src/sig.k8s.io/gateway-api
 COPY  . .
 ARG TAG
 ARG COMMIT
-RUN CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -a -o gateway-api-webhook \
+ARG ARCH
```
Comment (howardjohn) [inline Dockerfile:21]: can replace with https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

File: hack/build-and-push.sh [hunk 2]
```diff
@@ -64,10 +70,10 @@ fi
 
 # First, build the image, with the version info passed in.
 # Note that an image will *always* be built tagged with the GIT_TAG, so we know when it was built.
-docker build --build-arg COMMIT=${COMMIT} --build-arg TAG=${BINARY_TAG} \
-  			-t ${REGISTRY}/admission-server:${GIT_TAG} .
-docker push ${REGISTRY}/admission-server:${GIT_TAG}
+docker build --build-arg COMMIT=${COMMIT} --build-arg TAG=${BINARY_TAG} ARCH=${ARCH} \
```
Comment (howardjohn) [inline hack/build-and-push.sh:73]: I think we need to switch to `docker buildx build --platforms={...}`.

As-is, we won't get a single multi-arch image

Comment (akankshakumari393) [reply hack/build-and-push.sh:73]: Ok i will check👍

Comment (robscott) [reply hack/build-and-push.sh:73]: @akankshakumari393 are you still able to work on this? Anything we can help with?

**Comment by robscott** [comment]
It looks like this PR has gone stale and will be replaced by https://github.com/kubernetes-sigs/gateway-api/pull/1462, closing this one out.

**Pull request closed without merging**